Doesn't support actual amazon linux 2023.6+

Running on AWS t4g.nano Amazon Linux 2023 (kernel-6.12),

`/etc/os-release`:
```
NAME="Amazon Linux"
VERSION="2023"
ID="amzn"
ID_LIKE="fedora"
VERSION_ID="2023"
PLATFORM_ID="platform:al2023"
PRETTY_NAME="Amazon Linux 2023.10.20260105"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2023"
HOME_URL="https://aws.amazon.com/linux/amazon-linux-2023/"
DOCUMENTATION_URL="https://docs.aws.amazon.com/linux/"
SUPPORT_URL="https://aws.amazon.com/premiumsupport/"
BUG_REPORT_URL="https://github.com/amazonlinux/amazon-linux-2023"
VENDOR_NAME="AWS"
VENDOR_URL="https://aws.amazon.com/"
SUPPORT_END="2029-06-30"
```

the script output:
```
./openvpn-install.sh interactive
[INFO] The script only supports Amazon Linux 2023.6+
[INFO] Amazon Linux 2 is EOL and no longer supported.
[ERROR] Your version of Amazon Linux is not supported.
        Check the log file for details: openvpn-install.log
```

--- a/openvpn-install.sh
+++ b/openvpn-install.sh
@@ -1682,7 +1682,7 @@ function checkOS() {
 			log_fatal "Your version is not supported."
 		fi
 		if [[ $ID == "amzn" ]]; then
-			if [[ "$(echo "$PRETTY_NAME" | cut -c 1-18)" == "Amazon Linux 2023." ]] && [[ "$(echo "$PRETTY_NAME" | cut -c 19)" -ge 6 ]]; then
+			if [[ "$PRETTY_NAME" =~ ^Amazon\ Linux\ 2023\.([0-9]+) ]] && [[ "${BASH_REMATCH[1]}" -ge 6 ]]; then
 				OS="amzn2023"
 			else
 				log_info "The script only supports Amazon Linux 2023.6+"

